File: golang/007.go
package main

import "fmt"

func rotateArr(arr []int, positions int) []int {
	realPointer := 0
	copyArr := make([]int, len(arr))
	copy(copyArr,arr)

	fmt.Println(copyArr)

	for i := positions +1; i < len(arr); i++ {
		fmt.Print("here")
		arr[realPointer] = arr[i]
		realPointer++
	}
	fmt.Println(arr)



	for i := 0; i <= positions; i++ {
		arr[realPointer] = copyArr[i]
		realPointer++
	}

	return arr
}

// func main() {
// 	fmt.Println(rotateArr([]int{1,2,3,4,5,6,7}, 3))
// }

File: python/007.py
def rotateArr(arr, positions):
    realPointer = 0
    copyArr =  [x for x in arr]


    for x in range (positions+1, len(arr)):
        arr[realPointer] = arr[x]
        realPointer+=1
    
    for x in range (0, positions+1):
        arr[realPointer] = copyArr[x]
        realPointer+=1

    return arr

print(rotateArr([3,4,5,6,7,8,9,10],3))